### PR TITLE
chore(sqlparser): add do-apply-parser-test

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

```
./risedev do-apply-parser-test
```

`serde_yaml` doesn't support pretty-print, we can consider refactor the testdata files to toml later.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/sqlparser/test_runner/sqlparser_test.toml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    cat "$f" >> temp.apply.yaml
+    mv temp.apply.yaml "$SOURCE"
+  fi
+done
+
+rm *.apply.yaml
+
+echo "$(tput setaf 2)Diff applied!$(tput sgr 0)"
+'''
+category = "RiseDev - Test"
```
Comment (TennyZhuang) [inline src/sqlparser/test_runner/sqlparser_test.toml:55]:
```suggestion
category = "RiseDev - Test"

```

File: src/sqlparser/tests/testdata/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+*.apply.yaml
```
Comment (TennyZhuang) [inline src/sqlparser/tests/testdata/.gitignore:1]:
```suggestion
*.apply.yaml

```

**Review by xxchan — APPROVED** [review]
> serde_yaml doesn't support pretty-print


https://github.com/dtolnay/serde-yaml/issues/226#issuecomment-1198665455 How pretty do you want? 👀

**Comment by TennyZhuang** [comment]
> > serde_yaml doesn't support pretty-print
> 
> [dtolnay/serde-yaml#226 (comment)](https://github.com/dtolnay/serde-yaml/issues/226#issuecomment-1198665455) How pretty do you want? 👀

https://github.com/risinglightdb/pretty-xmlish

**Comment by xxchan** [comment]
.. So you want an alternative implemenation for pretty AST? I think that's irrelavent to yaml or toml?

**Comment by TennyZhuang** [comment]
> .. So you want an alternative implemenation for pretty AST? I think that's irrelavent to yaml or toml?

yaml should also break the string for human-readable.

**Pull request merged**